include/seqan3/range/container/aligned_sequence_adaptor_constant_access.hpp

```diff
@@ -0,0 +1,688 @@
+    const_iterator cbegin() const noexcept
+    {
+        return const_iterator{*this, 0};
+    }
+
+    //!\brief Return const iterator pointing to past-the-end element of underlying sequence.
+    const_iterator cend() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+     //!\}
+
+    /*!\name Boolean operators
+     * \{
+     */
+    /*!\brief Equality operator for aligned sequences.
+    *
+    * Two aligned sequences are the same if their literal sequences and gap
+    * positions are the same.
+    */
+    bool operator==(aligned_sequence_t const & rhs) const
+    {
+        return sequence == rhs.sequence && gap_vector == rhs.gap_vector;
+    }
+
+    //!\brief Unequality operator for aligned sequences.
+    bool operator!=(aligned_sequence_t const & rhs) const
+    {
+        return !(*this == rhs);
+    }
+
+    //!\brief Swap two aligned sequences and their support structures.
+    void swap(aligned_sequence_t & rhs)
+    {
+        sequence.swap(rhs.sequence);
+        std::swap(gap_vector, rhs.gap_vector);
+        rs.set_vector(&gap_vector);
+        rhs.rs.set_vector(&rhs.gap_vector);
+        ss.set_vector(&gap_vector);
+        rhs.ss.set_vector(&gap_vector);
+    }
+
+    //!\brief Return gapped sequence length.
+    size_type size() const noexcept
+    {
+        return gap_vector.size();
+    }
+
+    /*!\brief Return the maximal aligned sequence length.
+    *
+    * The maximal sequence length is limited by either the maximal size of the
+    * compressed sequence or the gap vector.
+    */
+    size_type max_size() const
+    {
+        return std::min<size_type>(sequence.max_size(), gap_vector.max_size());
+    }
+
+    //!\brief An aligned sequence is empty if it contains no alphabet letters or gaps.
+    bool empty() const
+    {
+        return sequence.empty() && gap_vector.size() == 0;
+    }
+    //!\}
+
+    /*!\name Assignment
+     * \{
+    */
+    //!\brief Assignment via iterators.
+    void assign(iterator it1, iterator it2) // these are aligned_seq iterators!
+    {
+        sequence.resize(0);
+        gap_vector = sdsl::bit_vector(it2 - it1, 0);
+        size_type i = 0;
+        for (; it1 != it2; ++it1)
+        {
+            assert(i < this->size());
+            if (*it1 == seqan3::gap::GAP)
+                gap_vector[i] = 1;
+            else
+                sequence.push_back(*it1);
+            ++i;
+        }
+        update_support_structures();
+    }
+
+    //!\brief Assignment via gapped alphabet sequence.
+    void assign(inner_type const sequence_)
+    {
+        sequence.resize(0);
+        gap_vector = sdsl::bit_vector(sequence_.size(), 0);
+        for (size_type i = 0; i < gap_vector.size(); ++i)
+        {
+            if (sequence_[i] == gap::GAP)
+                gap_vector[i] = 1;
+            else
+                sequence.push_back(sequence_[i]);
+        }
+        update_support_structures();
+    }
```

Comment:
try to re-use more code here, e.g. implement the interface with the two iterators and then in the other interface just call assign(ranges::begin(sequence), ranges::end(sequence));